src/expmv_fun.jl

```diff
@@ -41,22 +41,22 @@ function expmv(t, A, b; M = [], prec = "double", shift = false, full_term = fals
     end
     
     if isempty(M)
-        tt = 1
+        tt = 1.0
         M = select_taylor_degree(t * A, size(b, 2))
     else
         tt = t
     end
     
-    tol =   
-      if prec == "double"
-          2.0^(-53)
+    tol =
+      if prec == "half"
```

Comment:
I have a working branch that tries to address these style issues (and also implements proper 1 norm estimation). Since so much code is shared, I am still not dispatching on the type, but rather branching on the type (instead of having precision specified as a argument, I take precision to be implicitly specified based on use of `Float64` vs `Float32` etc). It is probably best to move the discussion to that branch's code -- `normest1`.